packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -380,23 +379,23 @@ class IOSSimulator extends Device {
     //         targeted applications cannot be run (even though the Flutter
     //         runner on the simulator is completely different).
 
-    RegExp versionExp = new RegExp(r'iPhone ([0-9])+');
-    Match match = versionExp.firstMatch(name);
-
-    // Not an iPhone. All available non-iPhone simulators are compatible.
-    if (match == null)
-      return true;
-
-    // iPhones 6 and above are always fine.
-    if (int.parse(match.group(1)) > 5)
-      return true;
+    // Check for unsupported iPads.
+    Match iPadMatch = new RegExp(r'iPad (2|Retina)', caseSensitive: false).firstMatch(name);
+    if (iPadMatch != null) {
+      _supportMessage = 'Flutter does not yet support iPad 2 or iPad Retina. Select an iPad Air or above.';
+      return false;
+    }
 
-    // The 's' subtype of 5 is compatible.
-    if (name.contains('iPhone 5s'))
-      return true;
+    // Check for unsupported iPhones.
+    Match iPhoneMatch = new RegExp(r'iPhone [0-5]').firstMatch(name);
+    if (iPhoneMatch != null) {
+      if (name == 'iPhone 5s')
+        return true;
+      _supportMessage = 'Flutter does not support yet iPhone 5 or earlier. Select an iPhone 5s or above.';
```

Comment:
That's what I get for being overzealous with vim 'redo command'. Fixed in #7859 along with the Mac host msg.